site/en/blog/chromium-chronicle-25/index.md

```diff
@@ -0,0 +1,83 @@
+---
+title: "The Chromium Chronicle #25: Thread Safety Annotations"
+description: >
+  Learn how Clang's static analysis framework takes over the toil of thread-safety proofs.
+layout: 'layouts/blog-post.njk'
+date: 2021-09-27
+hero: 'image/0g2WvpbGRGdVs0aAPc6ObG7gkud2/hgu6uTktp2ipmuODZZhP.jpg'
+alt: >
+  Chromium Chronicle image
+tags:
+  - chromium-chronicle
+---
+
+**Episode 25:** by Victor Constan in SFO (September, 2021)<br>
+[Previous episodes](/tags/chromium-chronicle/)
+
+In C++, ruling out the possibility of data races comes down to a small thread-safety correctness proof for every data member access.
+These proofs add up to a lot of mental toil, especially when reviewing or refactoring code.
+**Clang's static analysis framework takes over the toil of thread-safety proofs**.
+
+## Add `GUARDED_BY_CONTEXT()` to data members in thread-unsafe classes
+
+Most Chrome classes are thread-unsafe, and should be used on a single sequence.
+Add annotations to all data members that are not thread-safe.
```

Comment:
Practically, we're not going to annotate all members in all classes.  Consider guidance for when a class should have annotations.  e.g. "Annotate classes that are used in multithreading circumstances.  Add annotations to all data members..."

Reply:
I'd prefer that we keep this TotT short and sweet. 

I don't think we have consensus yet on what would get annotated, and I'd like to get this nudge out to folks as soon as possible. I am asking folks to [annotate *all* classes in storage code](https://docs.google.com/document/d/1QRIPkrOKU-QdDOpT4nak9UfMT8SKAzZfbe13fBah7ow/edit#heading=h.8ffdyerzmckm).

Reply:
Do we need to change anything based on this comment?

Reply:
The article does not need to change, but I stand by the suggestion.  I think readers will be confused when to do this.